### PR TITLE
[drci] don't mark jobs with an overly generic error as flaky

### PR DESCRIPTION
**Pull request by PaliC**
This should deal with https://github.com/pytorch/test-infra/issues/4927

### CODE REVIEW

File: torchci/lib/drciUtils.ts [hunk 1]
```diff
@@ -53,6 +53,12 @@ export const EXCLUDED_FROM_FLAKINESS = [
 // it increases the risk of getting misclassification. This guardrail can
 // be relaxed once we achieve better accuracy from the log classifier. This
 // sets the limit to 7 days
+
+export const ErrorsToNotDisable: RegExp[] = [
```
Comment (clee2000) [inline torchci/lib/drciUtils.ts:57]:
```suggestion
export const LINE_CAPTURES_EXCLUDED_FROM_FLAKINESS: RegExp[] = [
```
Rename to something more specific?

**Review by clee2000 — APPROVED** [review]
lgtm

File: torchci/lib/drciUtils.ts (hunk 1 above)
Comment (huydhn) [inline torchci/lib/drciUtils.ts:57]: A bit more context: When searching for similar failures, Dr.CI not only compares the capture line, but also the failure context of 12 commands before https://github.com/pytorch/test-infra/blob/main/aws/lambda/log-classifier/src/main.rs#L19.  So, a cancelled job is marked as flaky only when it matches with another exact canceled job from trunk.  I guess it's pretty rate, so we don't see it often till now.

File: torchci/lib/drciUtils.ts [hunk 2]
```diff
@@ -323,6 +329,13 @@ export async function hasSimilarFailures(
     if (!isEligibleCommit) {
       continue;
     }
+    if (
```
Comment (huydhn) [inline torchci/lib/drciUtils.ts:332]: I realize that let's not do the check here as it wastes a query to ElasticSearch.  You can add this check right at the begin of the function around line 258.  The failure string for the job is available there as `job.failure_captures`.

**Pull request merged**